### ISSUE TITLE
Expose SSE retry time updates via the public API

### ISSUE THREAD

**Issue by jamesmissen** [square/okhttp#8202]
Issue #5471 was closed with the recommendation to reconnect to SSE streams in the application layer.

It would be helpful to expose the retry times in [retry fields](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#retry) from the event stream, so that any reconnection attempts in a higher layer could respect the latest retry time if desired.

Currently, the retry times are parsed but they remain in the internal API only, terminating here:

https://github.com/square/okhttp/blob/aad8a1420d8f22a35138b92d7ecc14efc96155fe/okhttp-sse/src/main/java/okhttp3/internal/sse/RealEventSource.kt#L105

**Comment by yschimke** [comment]
Yep, this seems like a valid part of the API.  Exposed in other frameworks like https://github.com/spring-projects/spring-framework/blob/main/spring-web/src/main/java/org/springframework/http/codec/ServerSentEvent.java

And as you say, web EventSource implements retry observing that. https://html.spec.whatwg.org/multipage/server-sent-events.html#event-stream-interpretation

@swankjesse @JakeWharton any objection to exposing?  It would allow building a correct retrying implementation on top.

**Comment by JakeWharton** [comment]
Let's expose. We don't support automatic retry so we need to surface this to the caller.

**Comment by ricardomgarciaf** [comment]
Has there been any update regarding this? It'd be very useful to have this field exposed.

**Comment by JakeWharton** [comment]
No updates
